overlord/snapstate/handlers.go

```diff
@@ -1250,6 +1244,38 @@ func (m *SnapManager) undoCopySnapData(t *state.Task, _ *tomb.Tomb) error {
 	return nil
 }
 
+// writeMigrationStatus is used to re-write the state and sequence file (if they
+// exist). This must be called after the migration undo procedure is done
+// since it only then we know the actual final state of the migration.
+func writeMigrationStatus(t *state.Task, snapst *SnapState, snapName string) error {
+	st := t.State()
+
+	st.Lock()
+	err := Get(st, snapName, &SnapState{})
+	st.Unlock()
+	if err != nil && err != state.ErrNoState {
+		return err
+
+	}
+
+	if err == nil {
+		// migration state might've been written in the change; update it after undo
+		st.Lock()
+		Set(st, snapName, snapst)
```

Comment:
Shouldn't we return an error if this fails? (if not, a comment explaining why this is not needed would be good)

Reply:
`Set` doesn't return an error. Or do you mean `err`? Because that's returned above (except if it's a state.ErrNoState)

Reply:
Maybe small comment before the `if err == nil ` like `// snap has a snapstate (no ErrNoState error returned) so needs update`